test/widgets/app_test.dart

```diff
@@ -166,5 +167,49 @@ void main() {
       check(ZulipApp.scaffoldMessenger).isNotNull();
       check(ZulipApp.ready).value.isTrue();
     });
+
+    testWidgets('reportErrorToUserBriefly shows snack bar', (tester) async {
+      addTearDown(testBinding.reset);
+      await tester.pumpWidget(const ZulipApp());
+
+      final finder = find.descendant(
+        of: find.byType(SnackBar),
+        matching: find.text('test error message'));
+
+      // Prior to app startup, reportErrorToUserBriefly only logs.
+      reportErrorToUserBriefly('test error message');
+      check(ZulipApp.ready).value.isFalse();
+      await tester.pump();
+      check(takeLastReportedError()).equals('test error message');
+      check(finder.evaluate()).isEmpty();
+
+      check(ZulipApp.ready).value.isTrue();
+      // After app startup, reportErrorToUserBriefly displays a snack bar.
+      reportErrorToUserBriefly('test error message');
+      await tester.pump();
+      check(finder.evaluate()).single;
+    });
+
+    testWidgets('reportErrorToUserInDialog shows dialog', (tester) async {
+      addTearDown(testBinding.reset);
+      await tester.pumpWidget(const ZulipApp());
+
+      final finder = find.descendant(
+        of: find.byType(AlertDialog),
+        matching: find.text('test error message'));
+
+      // Prior to app startup, reportErrorToUserInDialog only logs.
+      reportErrorToUserInDialog('test error message');
+      check(ZulipApp.ready).value.isFalse();
+      await tester.pump();
+      check(takeLastReportedError()).equals('test error message');
+      check(finder.evaluate()).isEmpty();
+
+      check(ZulipApp.ready).value.isTrue();
+      // After app startup, reportErrorToUserInDialog displays a dialog.
+      reportErrorToUserInDialog('test error message');
+      await tester.pump();
+      check(finder.evaluate()).single;
```

Comment:
Could maybe be strengthened with `checkErrorDialog`; just a thought.

Reply:
Could be a bit repetitive, but it should be good to also check the title.